frontend/src/views/Dashboard/components/cards/idir/TransactionsCard.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { ROUTES } from '@/constants/routes'
+import { useCurrentUser } from '@/hooks/useCurrentUser'
+import {
+  useTransactionsTransfersInProgress,
+  useTransactionsInitiativeAgreementsInProgress,
+  useTransactionsAdminAdjustmentsInProgress
+} from '@/hooks/useTransactions'
+import { useNavigate } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import { Grid, List, ListItemButton } from '@mui/material'
+import BCWidgetCard from '@/components/BCWidgetCard/BCWidgetCard'
+import BCTypography from '@/components/BCTypography'
+import Loading from '@/components/Loading'
+
+const TransactionListItem = ({ text, route, number }) => {
+  const navigate = useNavigate()
+
+  return (
+    <Grid container alignItems="center" spacing={2}>
+      <Grid item xs={2}>
+        <BCTypography
+          variant="h3"
+          style={{ color: '#578260', textAlign: 'right' }}
+        >
+          {number}
+        </BCTypography>
+      </Grid>
+      <Grid item xs={10}>
+        <ListItemButton
+          component="div"
+          alignItems="flex-start"
+          onClick={() => navigate(route)}
+          sx={{ display: 'flex', alignItems: 'center', padding: '8px 16px' }}
+        >
+          <BCTypography
+            variant="subtitle2"
+            color="link"
+            sx={{
+              textDecoration: 'underline',
+              '&:hover': { color: 'info.main' },
+              display: 'flex',
+              alignItems: 'center'
+            }}
+          >
+            {text}
+          </BCTypography>
+        </ListItemButton>
+      </Grid>
+    </Grid>
+  )
+}
+
+export function TransactionsCard() {
+  const { t } = useTranslation(['common', 'org'])
+  const { data: currentUserData } = useCurrentUser()
+
+  const { data: transfersInProgressData, isLoading: transfersLoading } = useTransactionsTransfersInProgress()
```

Comment:
It would be better to have this component call only 1 endpoint rather than 3 at once. Could you refactor this to call an endpoint that returns all 3 counts together? A unioun repo call could possibly be used as well instead of 3 different queries, if those queries are not used elsewhere. Let me know if you want to discuss this in more detail. 

This query is used for the director dashboard widget just as a possible example.
```python
    @repo_handler
    async def get_director_review_counts(self):
        query = union_all(
            select(func.count().label('count'), literal('transfers').label('type'))
            .select_from(join(Transfer, TransferStatus, Transfer.current_status_id == TransferStatus.transfer_status_id))
            .where(TransferStatus.status == TransferStatusEnum.Recommended),

            select(func.count().label('count'), literal('compliance_reports').label('type'))
            .select_from(join(ComplianceReport, ComplianceReportStatus, ComplianceReport.status_id == ComplianceReportStatus.compliance_report_status_id))
            .where(ComplianceReportStatus.status == ComplianceReportStatusEnum.Recommended_by_manager),

            select(func.count().label('count'), literal('initiative_agreements').label('type'))
            .select_from(join(InitiativeAgreement, InitiativeAgreementStatus, InitiativeAgreement.current_status_id == InitiativeAgreementStatus.initiative_agreement_status_id))
            .where(InitiativeAgreementStatus.status == InitiativeAgreementStatusEnum.Recommended),

            select(func.count().label('count'), literal('admin_adjustments').label('type'))
            .select_from(join(AdminAdjustment, AdminAdjustmentStatus, AdminAdjustment.current_status_id == AdminAdjustmentStatus.admin_adjustment_status_id))
            .where(AdminAdjustmentStatus.status == AdminAdjustmentStatusEnum.Recommended)
        ).alias('counts')

        result = await self.db.execute(select(query.c.type, query.c.count))
        return dict(result.fetchall())

Reply:
Thanks for the suggestion. I'll apply the changes as part of the refactor ticket. Here are the refactor changes I have in mind so far:

- Using a materialized view for the counts
- One API for whole widget counts
- Moving the transactions texts into the dashboard.json locale file
- Passing frontend filter in navigate state from widget to transactions/reports list

Please let me know if you'd like me to apply these changes right away.